server/pkg/cache/cache.go

```diff
@@ -1,303 +1,557 @@
-	archivePath := c.getArchivePath(websiteAddress)
-	fileLock := c.newFileLock(archivePath)
-
-	err := readLockFile(fileLock)
-	if err != nil {
-		//nolint:wrapcheck
-		return time.Time{}, err
-	}
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 
-	defer lockUnlock(fileLock)
+	key := getHashKey(websiteAddress, fileName)
+	diskKey := getCacheKey(websiteAddress, fileName)
 
-	r, err := zip.OpenReader(archivePath)
-	if err != nil {
-		return time.Time{}, fmt.Errorf("failed to open zip archive %s: %v", archivePath, err)
+	// First try RAM cache without promoting
+	if value, ok := c.ramCache.Peek(key); ok {
+		if entry, ok := value.(*cacheEntry); ok {
+			return entry.modified, nil
+		}
 	}
 
-	defer closeReader(r)
+	// If not in RAM cache, try disk cache without promoting
+	var modified time.Time
+	err := c.db.View(func(txn *badger.Txn) error {
+		// Create the entry key prefix
+		entryKeyPrefix := make([]byte, 1+len(diskKey))
+		entryKeyPrefix[0] = entryTag
+		copy(entryKeyPrefix[1:], diskKey)
+
+		// Create new byte slice for timestamp key
+		timestampKey := make([]byte, len(entryKeyPrefix)+1)
+		copy(timestampKey, entryKeyPrefix)
+		timestampKey[len(entryKeyPrefix)] = entrySubTagTime
+
+		item, err := txn.Get(timestampKey)
+		if err == badger.ErrKeyNotFound {
+			return fmt.Errorf("timestamp not found for file %s in website %s", fileName, websiteAddress)
+		}
+		if err != nil {
+			return err
+		}
 
-	for _, f := range r.File {
-		if f.Name == fileName {
-			return f.Modified, nil
+		timestampValue, err := item.ValueCopy(nil)
+		if err != nil {
+			return err
 		}
+
+		modified = time.Unix(0, int64(binary.BigEndian.Uint64(timestampValue)))
+		return nil
+	})
+
+	if err != nil {
+		return time.Time{}, err
 	}
 
-	return time.Time{}, fmt.Errorf("file %s not found in archive %s", fileName, archivePath)
+	return modified, nil
 }
 
 // Read returns the content of a resource in the cache for a given website
 func (c *Cache) Read(websiteAddress string, resourceName string) ([]byte, error) {
-	archivePath := c.getArchivePath(websiteAddress)
-	fileLock := c.newFileLock(archivePath)
-
-	err := readLockFile(fileLock)
-	if err != nil {
-		//nolint:wrapcheck
-		return nil, err
-	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
 
-	defer lockUnlock(fileLock)
+	key := getHashKey(websiteAddress, resourceName)
+	diskKey := getCacheKey(websiteAddress, resourceName)
 
-	r, err := zip.OpenReader(archivePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open zip archive %s: %v", archivePath, err)
+	// First try RAM cache and promote if found
+	if value, ok := c.ramCache.Get(key); ok {
+		if entry, ok := value.(*cacheEntry); ok {
+			return entry.content, nil
+		}
 	}
 
-	defer closeReader(r)
+	// If not in RAM cache, check disk cache
+	var content []byte
+	var timestamp []byte
+	err := c.db.Update(func(txn *badger.Txn) error {
+		// Create the entry key prefix
+		entryKeyPrefix := make([]byte, 1+len(diskKey))
+		entryKeyPrefix[0] = entryTag
+		copy(entryKeyPrefix[1:], diskKey)
```

Comment:
it seems you can factorize some code with `GetLastModified` function

Reply:
done